src/NNCache.h

```diff
@@ -29,6 +29,17 @@
 
 class NNCache {
 public:
+
+    /*
+    Maximum size of the cache in number of items.
+    */
+    static constexpr int MAX_CACHE_COUNT = 150'000;
```

Comment:
You need to define these in the .cpp file too or you'll get obscure compilation errors at some point. (What you have here is only a declaration).

Check the other constants in FastBoard.cpp/h if you don't immediately get what I mean.